### ISSUE TITLE
[HOLD for Payment 2024-10-04] [$250] [Search v2.4] [App] Introduce <SearchRouter> component

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#49118]
Implement [this section](https://docs.google.com/document/d/1Uf7UwMYTmek2qWe37SqkQZnOZYrnrfPucOLzXMv-3WM/edit#bookmark=id.by0r5glx29ja) of the doc

cc @Kicu @289Adam289 @Guccio163 @SzymczakJ

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021834650090913251873</li>
        <li>Upwork Job ID: 1834650090913251873</li>
        <li>Last Price Increase: 2024-09-13</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by Kicu** [comment]
hey, since i was preparing a draft POC while writing the doc, I have some code for this, please assign me

**Comment by twisterdotcom** [comment]
PR is going through deployment: https://github.com/Expensify/App/pull/48785

**Comment by Kicu** [comment]
also I might need to add a few more followups after my PR was merged, because we were trying to merge it ASAP so more tasks gets unblocked.
Now more than 1 person can work on search 2.4 at the same time, but I might have another PR related to this task. There are small bugs and tweaks

**Comment by luacmartins** [comment]
Sounds good. Let's work on the follow ups

**Comment by luacmartins** [comment]
@Kicu I think this issue is done, right? I believe we addressed all the follow up we needed to? Am I missing something?

**Comment by Kicu** [comment]
@luacmartins yes I believe so.
There will be some followups added to this, but they are done (by @adamgrzybowski) as part of this issue https://github.com/Expensify/App/pull/49462

(because the input component to type in query is shared between the Router and SearchResults page).

This one we can consider done.

**Comment by luacmartins** [comment]
Cool, I think only thing left is payment then.

**Comment by ikevin127** [comment]
⚠️ Automation failed here -> this should be on **[HOLD for Payment 2024-10-04]** according to yesterday’s production deploy from https://github.com/Expensify/App/pull/49379#issuecomment-2379772885.

@luacmartins If payment will happen here, https://github.com/Expensify/App/issues/49122 can be closed as it was addressed in the same PR.

**Comment by twisterdotcom** [comment]
Okay, I see @ikevin127 reviewed both https://github.com/Expensify/App/pull/49379 and https://github.com/Expensify/App/pull/48785 and hasn't been paid yet for either, so $500 makes sense here.

https://github.com/Expensify/App/pull/48785 was authored by @Kicu and https://github.com/Expensify/App/pull/49379 was authored by @luacmartins, so @Kicu should only get $250 and @luacmartins should only get a pat on the pack and my heartfelt thanks.

**Comment by twisterdotcom** [comment]
Neither of these jobs I think have hires on in Upwork, so I'll send some contracts now ahead of time. I also see @Kicu is from SWM, so you also get heartfelt thanks instead too.

@ikevin127 - $500 [here](https://www.upwork.com/nx/wm/offer/104233352)
@Kicu - my heartfelt thanks [here](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExcmwzZDFwbGxycDAyeTJ2bTFqaHJtdmdtbmVnNWxzaXZhYjl6NDNnbSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l3q2wJsC23ikJg9xe/giphy-downsized-large.gif)
@luacmartins - my heartfelt thanks [here](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExOHF1MzI2YWs0ZDVlbnNhNWVqbmkyYXNwbTM3bHVxODJjbW1zbzM5ZSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/fVtcfEXWQJQUbsF1sH/giphy.gif)

**Comment by ikevin127** [comment]
That's correct, offer accepted and a heartfelt thank you!

**Comment by luacmartins** [comment]
Thanks. Sending heartfelt thanks to all of you.
